### ISSUE TITLE
[Dashboard][Chart List View] The search is only performed on the charts that are already listed

### ISSUE THREAD

**Issue by ValentinC-BR** [apache/superset#13727]
The chart list seems to display ~100 charts.
When the users looks for a chart by typing keywords, the search is only performed on these 100 charts.

As a results, many charts cannot be found.

### Expected results

The search should be performed on the complete list of charts a user owns.

The limitation (100, or whatever the number is) should be applied on the search results, so that the user can find the desired charts

### Actual results

The search is performed on the filtered list.

#### Screenshots

/

#### How to reproduce the bug

1. Create 101 charts (not sure of the number)
2. Create a dashboard
3. Look for your 101th chart by using the search box
4. Error : your chart can't be found

### Environment

- superset version: 1.0.1
- python version: 3.7.9
- node.js version: doesn't apply, I run on Kubernetes, using gunicorn as server
- source : Athena

### Checklist

Make sure to follow these steps before submitting your issue - thank you!

- [ ] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [X] I have reproduced the issue with at least the latest released version of superset.
- [X] I have checked the issue tracker for the same issue and I haven't found one similar.

### Additional context

/

**Comment by junlincc** [comment]
Thanks for reporting?

@nytai are you aware that the limit of 100 items was added previsouly?

**Comment by nytai** [comment]
I am not aware of this limitation. The search should be against the backend and the entire table should be searched.

**Comment by nytai** [comment]
Ohh, @ValentinC-BR might be referring to the "add chart" widget in the dashboard view. The search might be happening locally there

**Comment by nytai** [comment]
Yup, looks like for that widget we just fetch the first 200 results and the search is just local to those results.

**Comment by junlincc** [comment]
@nytai thanks for checking! 

@ValentinC-BR I labeled it as feature request. It's not in our 2021-2022 roadmap but it would be really nice to have it implemented. I'm not sure how it will affect performance, but let us know if you would like to contribute and make the change.

**Comment by ValentinC-BR** [comment]
OK thanks.

Quite surprised it's a feature request, I thought it was a bug as it's not clearly stated that the search is performed on the charts already displayed, but anyway.

I'm an end-user so I won't be able to contribute more ! Let me know if this can come sooner than 2022. There are workarounds, and not all users are affected, this is just a (quite big in some cases) confort improvement ;-)
